code/addons/onboarding/src/Onboarding.tsx

```diff
@@ -3,15 +3,14 @@ import React, { useCallback, useEffect, useState } from 'react';
 import { SyntaxHighlighter } from 'storybook/internal/components';
 import { SAVE_STORY_RESPONSE } from 'storybook/internal/core-events';
 
-import type { Step } from 'react-joyride';
 import { type API } from 'storybook/manager-api';
 import { ThemeProvider, convert, styled, themes } from 'storybook/theming';
 
+import { HighlightElement } from '../../../core/src/manager/components/TourGuide/HighlightElement';
+import { TourGuide } from '../../../core/src/manager/components/TourGuide/TourGuide';
```

Comment:
I get scared when I see this. We should use relative imports only when unavoidable. Suppose the `HighlightElement` imports a package that is not listed in `addon-onboarding` 's' package.json'. In that case, things will crash tremendously (or get pre-bundled, which I'm not sure, potentially leading to duplicates at runtime). The side effect of this relative import is unpredictable. I think moving the components to `manager` was a mistake.

Reply:
> things will crash tremendously

They won't.

> or get pre-bundled

That's what will happen.

> potentially leading to duplicates at runtime

Never duplicates of anything from core, which is what matters.

> The side effect of this relative import is unpredictable

This needs clarification, what are the possibilities that you're considering?

> I think moving the components to manager was a mistake.

I think this is better than putting new component is `core/src/component`, where they will have to adhere to semver.

The only way to prevent future traps of people using things they shouldn't be using is by not exposing them.